src/annotator/config/index.js

```diff
@@ -11,30 +11,23 @@ var sharedSettings = require('../../shared/settings');
 function configFrom(window_) {
   var config = {
     app: settings.app(window_.document),
+
+    // Extract the default annotation ID or query from the URL.
+    //
+    // The Chrome extension or proxy may already have provided this config via
+    // a tag injected into the DOM, which avoids the problem where the page's
+    // JS rewrites the URL before Hypothesis loads.
+    //
+    // In environments where the config has not been injected into the DOM,
+    // we try to retrieve it from the URL here.
```

Comment:
What actually happens is that, if the Chrome extension has moved the query or annotations setting into a js-hypothesis-config object in the page, then the `Object.assign` below will read the setting(s) from the page and overwrite the (possibly `null`) setting values already set from the URL.

I've left this comment here for now. In the future the fact that query and annotations comes from the URL or from a js-hypothesis-config will be encapsulated within the `settings.query()` and `settings.annotations()` functions and can be docstringed and unit-tested there.